tests/e2e/e2e_deployment_client_server_test.go

```diff
@@ -0,0 +1,171 @@
+				svcAccDef, deploymentDef, svcDef = td.SimpleDeploymentApp(
+					SimpleDeploymentAppDef{
+						name:         srcClient,
+						namespace:    srcClient,
+						replicaCount: int32(replicaSetPerService),
+						command:      []string{"/bin/bash", "-c", "--"},
+						args:         []string{"while true; do sleep 30; done;"},
+						image:        "songrgg/alpine-debug",
+						ports:        []int{80}, // Can't deploy services with empty/no ports
+					})
+				_, err = td.CreateServiceAccount(srcClient, &svcAccDef)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = td.CreateDeployment(srcClient, deploymentDef)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = td.CreateService(srcClient, svcDef)
+				Expect(err).NotTo(HaveOccurred())
+
+				wg.Add(1)
+				go func(wg *sync.WaitGroup, srcClient string) {
+					defer wg.Done()
+					Expect(td.WaitForPodsRunningReady(srcClient, 200*time.Second, replicaSetPerService)).To(Succeed())
+				}(&wg, srcClient)
+			}
+			wg.Wait()
+
+			// Create traffic rules
+			// Deploy allow rules (10x)clients -> server
+			for _, srcClient := range sourceNamespaces {
+				httpRG, trafficTarget := td.CreateSimpleAllowPolicy(
+					SimpleAllowPolicy{
+						RouteGroupName:    srcClient,
+						TrafficTargetName: srcClient,
+
+						SourceNamespace:      srcClient,
+						SourceSVCAccountName: srcClient,
+
+						DestinationNamespace:      destApp,
+						DestinationSvcAccountName: destApp,
+					})
+
+				// Configs have to be put into a monitored NS, and osm-system can't be by cli
+				_, err = td.CreateHTTPRouteGroup(srcClient, httpRG)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = td.CreateTrafficTarget(srcClient, trafficTarget)
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			// Create Multiple HTTP request structure and fill it with appropriate details
+			requests := HTTPMultipleRequest{
+				Sources: []HTTPRequestDef{},
+			}
+			for _, ns := range sourceNamespaces {
+				pods, err := td.client.CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{})
+				Expect(err).To(BeNil())
+
+				for _, pod := range pods.Items {
+					requests.Sources = append(requests.Sources, HTTPRequestDef{
+						SourceNs:        ns,
+						SourcePod:       pod.Name,
+						SourceContainer: ns, // container_name == NS for this test
+
+						Destination: fmt.Sprintf("%s.%s", destApp, destApp),
+
+						HTTPUrl: "/",
+						Port:    80,
+					})
+				}
+			}
+
+			var results HTTPMultipleResults
+			success := td.WaitForRepeatedSuccess(func() bool {
+				// Issue all calls, get results
+				results = td.MultipleHTTPRequest(&requests)
+
+				// Care, depending on variables there could be a lot of results
+				td.PrettyPrintHTTPResults(&results)
+
+				// Verify success conditions
+				for _, ns := range results {
+					for _, podResult := range ns {
+						if podResult.Err != nil || podResult.StatusCode != 200 {
+							return false
+						}
+					}
+				}
+				return true
+			}, 5, 150*time.Second)
```

Comment:
Could you declare the values here as a constant or at least a variable or with inline comments, so that it can understandable.

Reply:
> Please open an issue to track this.
Have a list if things on my whiteboard. Going to take care of it today.